src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,34 +39,43 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        String [] addressArray = address.split(",");
+        this.addressBlock = new Block(addressArray[BLOCK_INDEX]);
```

Comment:
The way you split here and the one in the `isValidAddress` is not consistent, better make it consistent.